src/chainparams.cpp

```diff
@@ -376,6 +380,8 @@ class CRegTestParams : public CTestNetParams
         nTargetSpacing = 1 * 60;        // PIVX: 1 minutes
         bnProofOfWorkLimit = ~uint256(0) >> 1;
         nLastPOWBlock = 250;
+        nPivxBadBlockTime = 1489001494;  // Regtest shouldn't apply, so we will just use testnet settings since they shouldn't match
+        nPivxBadBlocknBits = 0x1e0a20bd; // Regtest shouldn't apply, so we will just use testnet settings since they shouldn't match
```

Comment:
Regtest is started on demand (with the latest code) and destroyed when the test is over. So it will never need any kind of escaping for bad blocks.

Furthermore `CRegTestParams` is a subclass of `CTestNetParams` so we have called the parent constructor (and set those values for `nPivxBadBlockTime` and `nPivxBadBlocknBits`) already here.

Reply:
> Regtest is started on demand (with the latest code) and destroyed when the test is over. So it will never need any kind of escaping for bad blocks.

However, if there was a bad nBit during regtest; it would go into the code path that checks if it's because it's the bad block.  That's what I was trying to cover... (how to ensure the code path to return true wasn't entered on regtest) but as you further explain; it's covered by the inheritance.

Reply:
fixed up; and cleaned up my rebase mess too